Add RegSearchSheet

r-uchino opened the pull request:
Add a regular expression to the search method

Since the return value is returned by MAP,
 it is convenient when you want to search by regular expression and obtain multiple cell coordinates

PS:
I am using translation, I'm sorry in dirty English :(

xuri commented:
Hi @r-uchino, thanks for your PR. Please read [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md) and add unit tests for your changes. I've left some comments.

xuri commented on sheet.go:
```diff
@@ -707,6 +708,59 @@ func (f *File) SearchSheet(sheet, value string) []string {
+//    xlsx.RegSearchSheet("Sheet1", "[0-9]")
+//
+// return map[CELL:result CELL:result ...]
+func (f *File) RegSearchSheet(sheet, value string) map[string]string {
```
I suggest that the return value of the function `RegSearchSheet` use type `[][]string` instead of `map[string]string` will be better, for example: `[][]string{"B3", "cell value"}`.

xuri commented on sheet.go:
```diff
@@ -707,6 +708,59 @@ func (f *File) SearchSheet(sheet, value string) []string {
+				_ = decoder.DecodeElement(&r, &startElement)
+				for _, colCell := range r.C {
+					val, _ := colCell.getValueFrom(f, d)
+					_reg := regexp.MustCompile(value)
```
The contents of functions `SearchSheet` and `RegSearchSheet` are mostly the same, we can refactor function `SearchSheet` to add a variable length parameter `reg` instead of add the new function `RegSearchSheet`, and change return type of function `SearchSheet` to `[][]string`.

xuri commented on sheet.go:
```diff
@@ -707,6 +708,59 @@ func (f *File) SearchSheet(sheet, value string) []string {
+						continue
+					}
+					key := fmt.Sprintf("%s%d", strings.Map(letterOnlyMapF, colCell.R), r.R)
+					result[key] = _reg.FindString(val)
```
It'is not recommended to name variables with `_` at the start.

r-uchino commented:
# PR Details

`RegSearchSheet(sheet, value string)` method added

## Description

`RegSearchSheet(sheet, value string) ` Method added,It is similar to `SearchSheet(sheet, value string)`, but uses regular expressions for the search. Returns a cell that matches a regular expression


## Related Issue


## Motivation and Context

It is necessary to describe a certain tag in the Excel file and to identify and change the content from the program (for example, you can get the cell marked as {tag: thisPrice} Replace with variable values)
I implemented it for that

## How Has This Been Tested

https://github.com/360EntSecGroup-Skylar/excelize/pull/316/commits/6fba208b2cee4dcb3b3021f86255dc608e0d5f74

We added unit tests as well as existing methods

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

r-uchino commented:
I did not grasp accurately, I am sorry, I sent the change as described above, I am pleased if you can confirm, Thank you for taking care of polite
